### DISCUSSION TITLE
setup

### DISCUSSION THREAD

**Discussion by RVishalkumar** [community/community#90043] [New to GitHub]
$ git push -u origin master
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by khot-aditya** [comment]
Hey, did you try add your SSH key to account?

**Comment by NipulM** [comment]
The most common cause of this problem is an issue with your SSH keys.

Make sure you have the correct SSH key added to your GitHub account.
You can check that by type-in:
`cat ~/.ssh/id_rsa.pub
`

If your SSH key to SSH agent is not added
Run:
```
eval "$(ssh-agent -s)"
ssh-add ~/.ssh/id_rsa
```

if the URL is using HTTPS after check by doing `git remote -v`, you can change it to the SSH URL by:
`git remote set-url origin {url of your repo}
`

Hope this helps!
